crates/std_detect/src/detect/os/linux/aarch64.rs

```diff
@@ -6,6 +6,23 @@ use crate::detect::{bit, cache, Feature};
 /// Try to read the features from the auxiliary vector, and if that fails, try
 /// to read them from /proc/cpuinfo.
 pub(crate) fn detect_features() -> cache::Initializer {
+    #[cfg(all(target_arch = "aarch64", target_os = "android"))]
+    {
+        // Samsung Exynos 9810 has a bug that big and little cores have different
+        // ISAs. And on older Android (pre-9), the kernel incorrectly reports
+        // that features available only on some cores are available on all cores.
+        // https://reviews.llvm.org/D114523
+        let mut arch = [0_u8; libc::PROP_VALUE_MAX as usize];
+        let len = unsafe {
+            libc::__system_property_get(
+                b"ro.arch\0".as_ptr() as *const libc::c_char,
+                arch.as_mut_ptr() as *mut libc::c_char,
+            )
+        };
+        if len > 0 && arch.starts_with(b"exynos9810\0") {
+            return cache::Initializer::default();
```

Comment:
We still want to probe for v8.0 features such as aes/sha/crc32.

Reply:
IIUC this is a bit tricky because even the armv8.0 optional features may not be consistent between little and big.

So for now, all I can think of is one of the following

- Only check features that are known to be available on exynos-m3 (armv8.0 cores).
  (According to https://github.com/golang/go/issues/28431) The old kernel seems to return auxv/cpuinfo based on features available on cortex-a55 (armv8.2 cores), so it can properly handle features even if they are available on exynos-m3 but not on cortex-a55 (though such features probably do not exist).
- Continue detection on Android 9+ and return default value on pre-9.

The first one seems to make more sense, as it does not require additional __system_property_get to get the version and works on older kernels.

Reply:
Implemented the first approach in https://github.com/rust-lang/stdarch/pull/1378/commits/9ce03f137315630630bca6acb29145d5048fb6d2.